src/errors.ts

```diff
@@ -8,12 +8,22 @@ export const ResponseSerializationError = createError<[{ cause: Error }]>(
   500,
 );
 
+export type ResponseSerializationErrorType = FastifySchemaValidationError & {
+  cause: ZodError;
+  serialization: {
+    method: string;
+    url: string;
+  };
+};
+
 export const InvalidSchemaError = createError<[string]>(
   'FST_ERR_INVALID_SCHEMA',
   'Invalid schema passed: %s',
   500,
 );
 
+export type InvalidSchemaErrorType = FastifySchemaValidationError;
```

Comment:
`FastifySchemaValidationError` is a different error and should not be used here.

```
export interface FastifySchemaValidationError {
  keyword: string;
  instancePath: string;
  schemaPath: string;
  params: Record<string, unknown>;
  message?: string;
}
```